pkg/server/api.go

```diff
@@ -134,31 +134,47 @@ func (sh *APIHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusNotFound)
 		return
 	}
-	// we know we're at 3.2 in code.. serve directly, parsing is expensive...
-	// we're doing it during an HTTP request, and most notably before we write the HTTP headers
+
 	var serveConf *runtime.RawExtension
-	if reqConfigVer.Equal(*semver.New("3.2.0")) {
+	if reqConfigVer.Equal(*semver.New("3.3.0")) {
+		// If the config version is greater or equal to 3.3.0 we can add the kargs from the
+		// MachineConfig into the returned config
+		kargs, err := sh.server.GetKernelArguments(cr)
+		if err != nil {
+			w.Header().Set("Content-Length", "0")
+			w.WriteHeader(http.StatusInternalServerError)
+			glog.Errorf("couldn't get kernel arguments for v3.3.0 req: %v, error: %v", cr, err)
+			return
+		}
+		if err := ctrlcommon.AddV33IgnitionKernelArguments(conf, kargs); err != nil {
```

Comment:
So I could be wrong on this, but the bootstrap server and the regular in-cluster server both can make use of this Addition of Ignition Kargs, right?

What's preventing the commonly-seen bootstrap issue of drifting between configs? i.e. the initial bootstrap served config will have the karg snippet if its using ignition spec 3.3.0, but the actual in-cluster rendered master MC does not. Won't that cause a `rendered-master-xxx` not found error? Or is this transparent to that process/I am misunderstanding something.